draft/0006-pat.md

```diff
@@ -0,0 +1,127 @@
+
+## Motivation
+
+The current authentication flow in Logto is optimized for interactive user sessions. However, there are scenarios where users need to authenticate non-interactive processes or third-party integrations. These include:
+
+1. Automated scripts or bots that perform actions on behalf of the user.
+2. Integration with external tools and services that require API access.
+3. Long-running processes that outlast typical session durations.
+
+## Personal Access Token (PAT) Creation
+
+The client makes a post request to the tenant's Management API `/api/users/:userId/personal-access-tokens` to create a PAT for the user. The following parameters are included in the HTTP request entity-body using the `application/json` format.
+
+1. `name`: REQUIRED. A user-defined name for the PAT to identify its purpose, should be unique in the user's account.
+2. `expiresAt`: OPTIONAL. The epoch time in milliseconds when the token will expire. If not provided, the token will never expire.
+
+If the request to create a PAT is successful, the tenant's Management API returns the PAT in the response.
+
+1. `name`: REQUIRED. The name of the PAT.
+2. `value`: REQUIRED. The value of the PAT.
+3. `expiresAt`: OPTIONAL. The epoch time in milliseconds when the token will expire.
+
+### Example PAT creation
+
+```http
+POST /api/users/123/personal-access-tokens HTTP/1.1
+Host: tenant.logto.app
+Content-Type: application/json
+Authorization: Bearer <access-token-with-management-api-access>
+
+{
+  "name": "My PAT",
+}
+```
+
+```http
+HTTP/1.1 201 Created
+Content-Type: application/json
+
+{
+  "name": "My PAT",
+  "value": "pat_W51arOqe7nynW75nWhvYogyc",
+  "expiresAt": null
+}
+```
+
+## Token Exchange with PAT
+
+Extend the token exchange flow to accept PATs as subject tokens and exchange them for short-lived access tokens.
+
+### Request
+
+The client makes a token exchange request to the tenant's token endpoint with a special grant type using the HTTP POST method. The following parameters are included in the HTTP request entity-body using the `application/x-www-form-urlencoded` format.
+
+1. `grant_type`: REQUIRED. The value of this parameter must be `urn:ietf:params:oauth:grant-type:token-exchange` indicates that a token exchange is being performed.
+2. `resource`: OPTIONAL. The resource indicator, the same as other token requests.
+3. `scope`: OPTIONAL. The requested scopes, the same as other token requests.
+4. `subject_token`: REQUIRED. The user's PAT.
+5. `subject_token_type`: REQUIRED. The type of the security token provided in the `subject_token` parameter. The value of this parameter must be `urn:logto:token-type:personal-access-token`.
```

Comment:
should we use snake_case here? i mean `urn:logto:token_type:personal_access_token`

as we are using snake_case elsewhere (e.g. `urn:logto:scope:organization_roles`), and the only non-snake_case thing i saw in the Token Exchange RFC is `token-type`. even the following `access_token` is snake_case in `urn:ietf:params:oauth:token-type:access_token` which looks super weird

Reply:
Ok, let's keep the same format as `access_token`